gimp: fix all plugins build

avnik opened the pull request:
###### Motivation for this change

Some plugins was broken, and prevent build gimp with all plugins installed

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

avnik committed:
gimp: fix all plugins build

Mic92 commented on pkgs/applications/graphics/gimp/plugins/default.nix:
```diff
@@ -125,10 +127,11 @@ rec {
+    buildInputs = with pkgs; [ 
+      fftw 
+      autoreconfHook
+      intltool
```
autoreconfHook and intltool belong to `nativeBuildInputs`

avnik replied:
intltool is enough common for most plugins, should it be default?

Mic92 replied:
could be.

Mic92 commented on pkgs/applications/graphics/gimp/plugins/default.nix:
```diff
@@ -140,11 +143,11 @@ rec {
       url = mirror://sourceforge/gimp-texturize/texturize-2.1_src.tgz;
       sha256 = "0cdjq25g3yfxx6bzx6nid21kq659s1vl9id4wxyjs2dhcv229cg3";
     };
+    buildInputs = with pkgs; [ intltool perl ];
```
intltool belongs to nativeBuildInputs

Mic92 commented on pkgs/applications/graphics/gimp/plugins/default.nix:
```diff
@@ -156,7 +159,7 @@ rec {
     */
     name = "wavelet-sharpen-0.1.2";
-    buildInputs = [ gimp ] ++ gimp.nativeBuildInputs;
+    buildInputs = with pkgs; [ intltool ];
```
same here.

Mic92 commented on pkgs/applications/graphics/gimp/plugins/default.nix:
```diff
@@ -169,7 +172,7 @@ rec {
     */
     name = "lqr-plugin-0.6.1";
-    buildInputs = [ pkgconfig libLQR gimp ] ++ gimp.nativeBuildInputs;
+    buildInputs = with pkgs; [ intltool libLQR ];
```
same here.

Mic92 commented on pkgs/applications/graphics/gimp/plugins/default.nix:
```diff
@@ -23,7 +23,10 @@ let
-  // { name = "${a.name}-${gimp.name}-plugin"; }
+  // {
+      name = "${a.name}-${gimp.name}-plugin";
+      buildInputs = [ gimp gimp.gtk glib pkgconfig ] ++ (a.buildInputs or []);
```
`pkgconfig` belongs to nativeBuildInputs.

avnik replied:
Thinked about split  nativeBuildInputs, but was urge to rebuild my system. Will do.

avnik committed:
gimp: split buildInputs and nativeBuildInputs for plugins

avnik commented:
@Mic92  all stuff you requested was fixed

avnik committed:
gimp: promote intltool to default nativeBuildDepends

It used by gimp itself (so should be already present on machine builds
gimp+plugins) and 90% of plugins.

avnik commented:
@Mic92  also promoted intltool as default build input,  I think it ready to merge
